### DISCUSSION TITLE
Force TCP fragmentation

### DISCUSSION THREAD

**Discussion by danfoxley** [dannagle/PacketSender#310] [Q&A] [answered]
Hello,
Please help with my scenario. I'd like to force fragmentation by sending a large TCP packet and force the routers in between to fragment and re-assemble the packet on the remote side. Is it possible to send a packet larger than my MTU setting with PacketSender?

Thanks!

**Comment by dannagle** [comment, marked as answer]
The data transfer out the TCP socket is managed by your operating system, so the data will honor whatever MTU is configured. Packet Sender cannot override it. You will need to access a lower level library to do that.

Perhaps you could add a hop in between your destination with a very small MTU?

Something to keep in mind as you go about this test... many operating systems and routers invoke Nagle's Algorithm on TCP packets and will join packets back together to improve efficiency. You may not get the exact test you want unless you also configure that setting.

**Reply by danfoxley** [reply]
@dannagle Thanks. In case there is new insight here. Our WebRTC application sends out a >1500 byte DTLS packet. I see it on the wire being fragmented, as expected.  The receiver was blocking this fragmented DTLS packet. Since no reply (UDP) or ICMP reply, was thinking to send a greater than >1500 byte TCP packet and see what hop (TTL) may have sent a RST.  This helped previously with a TCP issue. I was able to nail down a firewall two hops short of the receive that was inspecting our traffic and blocking.

https://osqa-ask.wireshark.org/questions/60445/tcp-rst-and-ttl/
